Magento 2.3 compatibility

gabrielqs opened the pull request.

dverkade committed:
Galillei/magento-sqs#9: Module is not compatible with Magento 2.3
 - Changed composer.json file in order to be compatible with Magento 2.3

dverkade committed:
Galillei/magento-sqs#9: Module is not compatible with Magento 2.3
 - Changed magento/framework version number to correct version for Magento 2.3.

dverkade committed:
Changed publisher configuration in the di.xml file to be compatible with Magento 2.3

dverkade committed:
- Added ConnectionTypeResolver
- Changed di.xml to add resolver to the Magento MessageQueue Resolver

dverkade committed:
- Changed typo in the ConnectionTypeResolved class.
- Changed di.xml configuration to use the ExchangeFactory class.
- Added ExchangeFactory class to for creating the Exchange instance.

dverkade committed:
Get the lastest dev package

dverkade committed:
Changed the configuration class for accepting the endpoint configuration value for creating the SQS connection.

dverkade committed:
Removed the queue.xml file which is not needed for Magento 2.3 compatibility.

dverkade committed:
Changed di.xml for consuming message queue.

dverkade committed:
Added QueueFactory class for instantiating queues.

dverkade committed:
Magento 2.3 compatibility, changed the name of the Psr Message because the package name was changed.

dverkade committed:
Fixed warning thrown when publishing messages.

dverkade committed:
- Changed config test to be compatible with Magento 2.3.
- Added tests for prefix and endpoint configuration values.

dverkade committed:
Made ConfigOptionsList test compatible with Magento 2.3

dverkade committed:
Added test for queue creation, queue purging and queue deletion.

dverkade committed:
Fixed typo with double ; on the end of the line.

dverkade committed:
Removed unnecessary references from the Topology test.

dverkade committed:
Added Exchange & Queue Test

gabrielqs commented on Test/Unit/Model/ExchangeTest.php:
```diff
@@ -0,0 +1,98 @@
+            ->method('push')
+            ->with($this->envelope);
+
+        $this->assertEquals($this->exchange->enqueue(self::TOPIC_NAME, $this->envelope), null);
```
@dverkade no need for the assert, we don't need the test to break if the enqueue method starts returning something

gabrielqs commented on Test/Unit/Model/QueueTest.php:
```diff
@@ -0,0 +1,127 @@
+    {
+        $this->objectManager = new ObjectManager($this);
+
+        $this->context = $this->getMockBuilder(\Enqueue\Sqs\SqsContext::class)
```
Please use aliases whenever possible, will make the code clearer and easier to maintain

gabrielqs commented on Test/Unit/Model/QueueTest.php:
```diff
@@ -0,0 +1,127 @@
+
+    public function testAcknowledge()
+    {
+        $this->sqsConfig->expects($this->exactly(3))
```
We can expect any here. The getConnection being called exactly 3 times is an implementation detail, we don't need to test for it, mocking the return value is enough

dverkade replied:
Changed this to any.

gabrielqs reviewed (commented):
@dverkade tests look great, thanks! added a few notes there, can you please take a look whenever possible?

wolzogen commented on Model/ConnectionTypeResolver.php:
```diff
@@ -0,0 +1,51 @@
+     */
+    public function getConnectionType($connectionName)
+    {
+        return in_array($connectionName, $this->sqsConnectionName) ? 'sqs' : null;
```
Use, please, so:
```php
return isset($this->sqsConnectionName[$connectionName]) ? 'sqs' : null;
```
It will be executed much faster, without overhead

dverkade replied:
Changed

dverkade replied:
I reviewed this again, but this: 

return isset(($this$this->->sqsConnectionNamesqsConnectionNam [$connectionName]) ? 'sqs' : null;

Is different then: 

return in_array($connectionName, $this->sqsConnectionName) ? 'sqs' : null;

The array sqsConnectionName holds the names of the connections as values of that array. So the in_array function which is used here is correct, which checks for the values. isset will check the array key and returns null in this case.

wolzogen commented on Model/Queue.php:
```diff
@@ -82,7 +82,7 @@ public function dequeue()
+            $envelope = $this->createEnvelop($message);
             return $envelope;
         }
         return null;
```
Write like this:
```php
return $message ? $this->createEnvelop($message) : null;
```

dverkade replied:
Thanks! I was just checking the code and saw the typo and did not check actual code. This is way better and changed it.

dverkade committed:
- Removed unused private variables.
- Changed code based on code review.

dverkade commented:
@gabrielqs can you review again and merge this to the master if everything looks good?

dverkade committed:
Queue names only use prefix when a prefix is set in the configuration. If no prefix is set, it will not append a "_" in front of the queue name any more.

dverkade committed:
Changed the string used to test the secret key. We used the same string for testing the access key, so I somehow the module returned the access key when retrieving the secret key the test would pass. By changing this we will ensure that the test fails when this happens.

dverkade committed:
Changed test to reflect changes in the prefix of the queue name. When no prefix is set it will no longer still prefix the queue name with an underscore "_".

dverkade committed:
- Created the ConnectionTypeResolver unit test
- Reverted some changes in the ConnectionTypeResolver class which where correct in the first place.

dverkade committed:
- Fixed issue with the Topology class where it was using the incorrect queue name.
- Changed test to reflect changes in the prefix of the queue name. When no prefix is set it will no longer still prefix the queue name with an underscore "_".

dverkade committed:
Changed protected to private according to Magento best practices.

dverkade committed:
Made the use of the bindings possible.

gabrielqs commented on Model/Exchange.php:
```diff
@@ -44,23 +56,37 @@ public function __construct(
+        }
+
+        foreach ($exchange->getBindings() as $binding) {
+            if ($binding->getTopic() == $topic &&
```
This if expression is complex enough to deserve it's own method

gabrielqs reviewed (commented):
Can't believe this was all we were missing

dverkade committed:
Created private function for the binding statement.
